corehq/ex-submodules/pillowtop/tests/test_elasticsearch.py

```diff
@@ -294,3 +302,191 @@ def test_conflict(self):
 
         # attempt to create the same doc twice shouldn't fail
         self._send_to_es_and_check(doc)
+
+
+@skipIf(settings.ELASTICSEARCH_MAJOR_VERSION != 7, 'Only applicable for ES7')
+@es_test
+class TestILM(SimpleTestCase):
+
+    def setUp(self):
+        self.es = get_es_new()
+        self.es_interface = ElasticsearchInterface(self.es)
+        self.index = XFORM_INDEX_INFO.index
+        self.alias = XFORM_INDEX_INFO.alias
+        XFORM_INDEX_INFO.ilm_config = prefix_for_tests(MAX_DOCS)
+        self.es.cluster.put_settings({
+            "persistent": {"indices.lifecycle.poll_interval": "1s"}
+        })
+
+        with trap_extra_setup(ConnectionError):
+            ensure_index_deleted(self.index)
+            initialize_index_and_mapping(self.es, XFORM_INDEX_INFO)
+
+    def tearDown(self):
+        XFORM_INDEX_INFO.ilm_config = ""
+        ensure_index_deleted(self.index)
+
+    def rollover(self):
+        # wait for ILM to kick in.
+        #   Is double the poll_interval to provide enough buffer for ILM process
+        time.sleep(2)
```

Comment:
is there not a way to poll ES to see when this is done rather than just sleeping?

Reply:
There isn't a way to do this, Simon. I had raised this issue in ES forum https://discuss.elastic.co/t/manually-do-an-ilm-poll-during-development-and-tests/251421/2

The other hacky way is to do manual rollover, I added a special test case with that method and a note recommending others to use manual rollover https://github.com/dimagi/commcare-hq/pull/28574/files#diff-846fa31c914001912c264b65c70cef170a3ac4e606c6e3598663759f29d7418eR487

Reply:
surely you can still poll to see when the new index is added?

Reply:
https://github.com/dimagi/commcare-hq/pull/28574/commits/01a01943568338e59c9c30b57c16ea44dd465f45

Reply:
@snopoke I had to undo this, as this is resulting in few errors that can't be avoided without waiting, I've added relevant comment with more explanation https://github.com/dimagi/commcare-hq/pull/28574/commits/b05b3455cea9f9299d6286a0c99dbb3773c6ae15